akka-cluster-sharding/src/main/scala/akka/cluster/sharding/SliceRangeShardAllocationStrategy.scala

```diff
@@ -0,0 +1,264 @@
+          val neighborSlices = neighborShards.map(_.toInt)
+          val alternative =
+            if (neighborSlices.min > slice)
+              findRegionWithLowerNeighbor(slice, maxShards, currentShardAllocations)
+            else if (neighborSlices.max < slice)
+              findRegionWithUpperNeighbor(slice, maxShards, currentShardAllocations)
+            else
+              None
+
+          val selectedRegion =
+            alternative match {
+              case Some(alternativeRegion) =>
+                if (neighborShards.size >= currentShardAllocations(alternativeRegion).size)
+                  regionWithNeighbor
+                else
+                  alternativeRegion
+              case None =>
+                regionWithNeighbor
+            }
+          Future.successful(selectedRegion)
+
+        } else {
+          Future.successful(regionWithNeighbor)
+        }
+      case None =>
+        Future.successful(allocateWithoutNeighbor(slice, currentShardAllocations))
+    }
+  }
+
+  private def allocateWithoutNeighbor(
+      slice: Int,
+      currentShardAllocations: Map[ActorRef, IndexedSeq[ShardId]]): ActorRef = {
+
+    // sort regions by member age because if a node is added (random address) we don't want rebalance more than necessary
+    val regionsByMbr = regionEntriesFor(currentShardAllocations).map(entry => entry.member -> entry.region).toMap
+    val regions = regionsByMbr.keysIterator.toIndexedSeq.sorted(Member.ageOrdering).map(regionsByMbr(_))
+
+    // first look for an empty region (e.g. rolling update case)
+    regions.reverse.collectFirst { case region if currentShardAllocations(region).isEmpty => region } match {
+      case Some(emptyRegion) => emptyRegion
+      case None =>
+        val rangeSize = NumberOfSlices / regions.size
+        val i = slice / rangeSize
+
+        if (i < regions.size) {
+          regions(i)
+        } else {
+          // This covers the rounding case for the last region, which we just distribute over all regions.
+          // May also happen if member for that region has been removed, but that should be a rare case.
+          regions(slice % regions.size)
+        }
+    }
+
+  }
+
+  override def rebalance(
+      currentShardAllocations: Map[ActorRef, IndexedSeq[ShardId]],
+      rebalanceInProgress: Set[ShardId]): Future[Set[ShardId]] = {
+    import math.max
+    import math.min
+
+    def limit(numberOfShards: Int): Int =
+      max(1, min((relativeLimit * numberOfShards).toInt, absoluteLimit))
+
+    if (rebalanceInProgress.nonEmpty) {
+      // one rebalance at a time
+      emptyRebalanceResult
+    } else {
+      val sortedRegionEntries = regionEntriesFor(currentShardAllocations).toVector.sorted(ShardSuitabilityOrdering)
+      if (!isAGoodTimeToRebalance(sortedRegionEntries)) {
+        emptyRebalanceResult
+      } else {
+        // this is the number of slices per region that we are aiming for
+        val overfill = 1
```

Comment:
Should it align with the overfill on allocation? (just the 1 in diff but seems strange)